addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js

```diff
@@ -0,0 +1,184 @@
+import ProgressPlaceHolder from '@aws-ee/base-ui/dist/parts/helpers/BasicProgressPlaceholder';
+
+// TODO: remove tempData
+const tempData = [
+  {
+    objectName: 'test-objectName',
+    projectId: 'test-Project',
+    workspaceId: 'workspaceId',
+    workspaceOwner: ['test-workspaceOwner1', 'test-workspaceOwner2'],
+    study: ['study1', 'study2'],
+    egressStatus: 'egressStatus',
+  },
+];
+class ScEnvironmentEgressStoreDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    runInAction(() => {
+      // A flag to indicate if egress request for this egress store is already submitted
+      this.egressStoreRequestSubmitted = this.getEgressStoreDetailStore().egressStoreStatus === 'submitted';
+    });
+  }
+
+  componentDidMount() {
+    const store = this.getEgressStoreDetailStore();
+    if (!isStoreReady(store)) {
+      swallowError(store.load());
+    }
+  }
+
+  getEgressStoreDetailStore() {
+    return this.props.scEnvironmentEgressStoreDetailStore;
+  }
+
+  handleSubmitEgressRequest = () => {
+    runInAction(() => {
+      this.egressStoreRequestSubmitted = !this.egressStoreRequestSubmitted;
+      console.log('egress request submitted');
+    });
+    // TODO: invoke store action to actually submit the request
+  };
+
+  render() {
+    const store = this.getEgressStoreDetailStore();
+    let content = null;
+
+    if (isStoreError(store)) {
+      content = <ErrorBox error={store.error} className="pt2 mb2" />;
+    } else if (isStoreLoading(store)) {
+      content = <ProgressPlaceHolder segmentCount={1} className="mt2 mb2" />;
+    } else if (isStoreEmpty(store)) {
+      content = this.renderEmpty();
+    } else if (isStoreNotEmpty(store)) {
+      content = this.renderEgressStoreTable();
+    } else {
+      content = null;
+    }
+    return <div className="fadeIn animated">{content}</div>;
+  }
+
+  renderConnections() {
+    return (
+      <Segment placeholder className="mt2 mb2">
+        <Header icon className="color-grey">
+          <Icon name="file outline" />
+          Empty Egress Store
+          <Header.Subheader>This workspace egress store is empty.</Header.Subheader>
+        </Header>
+      </Segment>
+    );
+  }
+
+  renderEmpty() {
+    return (
+      <Segment placeholder className="mt2 mb2">
+        <Header icon className="color-grey">
+          <Icon name="file outline" />
+          Empty Egress Store
+          <Header.Subheader>This workspace egress store is empty.</Header.Subheader>
+        </Header>
+      </Segment>
+    );
+  }
+
+  renderEgressStoreTable() {
+    const pageSize = 5;
+    const showPagination = tempData.length > pageSize;
+    const egressStoreRequestSubmitted = this.egressStoreRequestSubmitted;
+    return (
+      <Segment placeholder className="mt2 mb2">
+        <ReactTable
+          data={tempData}
+          defaultSorted={[{ id: 'objectName', desc: true }]}
+          showPagination={showPagination}
```

Comment:
Will the API handle pagination as well? I didn't see how we are passing the pagination input.

Reply:
Good point, since the backend code is not ready yet, I am not making them as paged yet. In the future PR, I will make changes after the backend is ready for this.